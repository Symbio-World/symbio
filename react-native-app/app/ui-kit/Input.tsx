import * as React from 'react'
import {
  View,
  TextInput,
  KeyboardTypeOptions,
  ReturnKeyType,
} from 'react-native'
import { t } from 'react-native-tailwindcss'

type Props = {
  value?: string
  onSubmit?: () => void
  onChange?: (text: string) => void
  autoFocus?: boolean
  keyboardType?: KeyboardTypeOptions
  returnKeyType?: ReturnKeyType
}

export const Input: React.FC<Props> = ({
  value,
  onSubmit,
  onChange,
  autoFocus,
  keyboardType,
  returnKeyType,
}) => (
  <View style={[t.flex1]}>
    <TextInput
      value={value}
      onChangeText={onChange}
      onSubmitEditing={onSubmit}
      returnKeyType={returnKeyType}
      keyboardType={keyboardType}
      autoFocus={autoFocus}
      style={[t.border, t.borderGray500, t.h12, t.roundedLg, t.p3, t.textBlack]}
    />
  </View>
)
