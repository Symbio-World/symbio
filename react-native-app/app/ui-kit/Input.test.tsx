import * as React from 'react'
import { TextInput } from 'react-native'
import { render, fireEvent } from 'react-native-testing-library'
import { Input } from './Input'

describe('Input', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Input />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('passess value to TextInput', () => {
    const value = 'value'
    const { getByType } = render(<Input value={value} />)
    expect(getByType(TextInput).props.value).toEqual(value)
  })

  it('triggers onChange', () => {
    const handleChange = jest.fn()
    const text = 'text'
    const { getByType } = render(<Input onChange={handleChange} />)
    fireEvent(getByType(TextInput), 'onChangeText', text)
    expect(handleChange).toHaveBeenCalledWith(text)
  })

  it('triggers onSubmit', () => {
    const handleSubmit = jest.fn()
    const { getByType } = render(<Input onSubmit={handleSubmit} />)
    fireEvent(getByType(TextInput), 'onSubmitEditing')
    expect(handleSubmit).toHaveBeenCalledTimes(1)
  })

  it('passess returnKeyType to TextInput', () => {
    const returnKeyType = 'send'
    const { getByType } = render(<Input returnKeyType={returnKeyType} />)
    expect(getByType(TextInput).props.returnKeyType).toEqual(returnKeyType)
  })

  it('passess keyboardType to TextInput', () => {
    const keyboardType = 'email-address'
    const { getByType } = render(<Input keyboardType={keyboardType} />)
    expect(getByType(TextInput).props.keyboardType).toEqual(keyboardType)
  })

  it('passess autoFocus to TextInput', () => {
    const autoFocus = true
    const { getByType } = render(<Input autoFocus={autoFocus} />)
    expect(getByType(TextInput).props.autoFocus).toEqual(autoFocus)
  })
})
