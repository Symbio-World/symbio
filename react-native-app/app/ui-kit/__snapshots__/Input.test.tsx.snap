// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderColor": "#a0aec0",
        },
        Object {
          "height": 48,
        },
        Object {
          "borderRadius": 8,
        },
        Object {
          "padding": 12,
        },
        Object {
          "color": "#000",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
